File: PRAKTIKUM OOP/2nd_Week/Kendaraan.cpp
#include "Kendaraan.hpp"
#include <iostream>
using namespace std;

Kendaraan:: Kendaraan(int no_Kendaraan, int tahun_keluaran, string merk) {
    this-> no_Kendaraan   = no_Kendaraan;
    this-> tahun_keluaran = tahun_keluaran;
    this-> merk           = merk;
}

Kendaraan:: Kendaraan() : Kendaraan(0,2020,"merci")  {}

void Kendaraan:: PrintInfo() {
    cout << "bermerk "    << this->merk           << " "
         << "keluaran tahun " << this->tahun_keluaran << " "
         << "dengan nomor "   << this->no_Kendaraan   << " ";  
}

File: PRAKTIKUM OOP/9th_week_tambahan/Kendaraan.java
abstract class Kendaraan{
    protected int no_kendaraan;
    protected String kategori;
    protected String merk;
    protected int tahun_keluaran;

    public Kendaraan(int no_kendaraan, String kategori, String merk, int tahun_keluaran) {
        this.no_kendaraan = no_kendaraan;
        this.kategori = kategori;
        this.merk = merk;
        this.tahun_keluaran = tahun_keluaran;
    }
    public Kendaraan() {
        this(0, "mobil", "XXX", 0);
    }
    public Kendaraan(Kendaraan other) {
        this.no_kendaraan = other.no_kendaraan;
        this.kategori = other.kategori;
        this.merk = other.merk;
        this.tahun_keluaran = other.tahun_keluaran;
    }

    public void PrintInfo() {
        System.out.print("Kendaraan " + this.kategori +" dengan nomor " + this.no_kendaraan + " ini dibuat tahun " + this.tahun_keluaran + " dan bermerk " + this.merk);
    }

    public abstract int BiayaSewa(int lamaSewa);
}
